ccd-config-generator/src/main/java/uk/gov/hmcts/ccd/sdk/generator/AuthorisationCaseFieldGenerator.java

```diff
@@ -110,9 +110,9 @@ public void write(
       for (Search<T, R> basket :
           Iterables.concat(config.getWorkBasketInputFields(),
               config.getWorkBasketResultFields())) {
-        for (SearchField<R> field : basket.getFields()) {
-          if (!fieldRolePermissions.contains(field.getId(), role)) {
-            fieldRolePermissions.put(field.getId(), role, Collections.singleton(Permission.R));
+        for (SearchField<R> searchField : basket.getFields()) {
+          if (searchField.availableToRole(role) && !fieldRolePermissions.contains(searchField.getId(), role)) {
```

Comment:
if I remove this new `searchField.availableToRole(role)` check the tests still pass - can we update the config/tests to cover this check?

Reply:
Thank you, I've updated the test CCD config to ensure the tests also fail when the check is removed for work basket input fields